### PR TITLE
feat: log http response body in debug mode

### PR DESCRIPTION
**Pull request by MayurGubrele**
- Adding functionality to log HTTP response body when log_level=debug
- Classes affected: AbstractHttpSink.java, SerializableHttpResponse.java, SerializableHttpResponseTest.java HttpSinkTest.java, PromSinkTest.java

### CODE REVIEW

File: src/main/java/io/odpf/firehose/sink/http/SerializableHttpResponse.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SerializableHttpResponse implements Serializable {
```
Comment (lavkesh) [inline src/main/java/io/odpf/firehose/sink/http/SerializableHttpResponse.java:15]: what is the purpose of this class? 
we can just call the debug method, if debug level is enabled.

File: src/main/java/io/odpf/firehose/sink/common/AbstractHttpSink.java [hunk 2]
```diff
@@ -85,6 +90,10 @@ private boolean shouldLogRequest(HttpResponse response) {
         return response == null || getRequestLogStatusCodeRanges().containsKey(response.getStatusLine().getStatusCode());
     }
 
+    private boolean shouldLogResponse(HttpResponse response) {
+        return response != null;
```
Comment (lavkesh) [inline src/main/java/io/odpf/firehose/sink/common/AbstractHttpSink.java:94]: shouldn't this return true if the debug level is enabled?

Comment (lavkesh) [reply src/main/java/io/odpf/firehose/sink/common/AbstractHttpSink.java:94]: we can add a method in instrumentation if not already there.
and call that
```
 if (instrumentation.isDebugEnabled()) {
    instrumentation.logDebug("");
 }
 ```

**Review by arujit — APPROVED** [review]

**Comment by arujit** [comment]
@MayurGubrele this branch has some conflicts right now. Can you please check and update things accordingly. Will merge post that.

**Comment by lavkesh** [comment]
Let's check the instrumentation.isDebugEnabled() before called the instrumentation.debug() as the parameter evaluation cost is relatively more.. It's a standard practice, please read it here. 
https://logging.apache.org/log4j/1.2/manual.html#Performance

Let's remove the SerializableHttpResponse class entirely, It's not needed.

**Review by lavkesh — CHANGES_REQUESTED** [review]
Let's check the instrumentation.isDebugEnabled() before calling the instrumentation.debug() as the parameter evaluation cost is relatively more.. It's a standard practice, please read it here.
https://logging.apache.org/log4j/1.2/manual.html#Performance

Let's remove the SerializableHttpResponse class entirely, It's not needed.

**Review by lavkesh — APPROVED** [review]

**Pull request merged**